packages/network-controller/src/NetworkController.ts

```diff
@@ -1253,108 +1980,77 @@ export class NetworkController extends BaseController<
    * @returns The populated network client registry.
    */
   #ensureAutoManagedNetworkClientRegistryPopulated(): AutoManagedNetworkClientRegistry {
-    const autoManagedNetworkClientRegistry =
-      this.#autoManagedNetworkClientRegistry ??
-      this.#createAutoManagedNetworkClientRegistry();
-    this.#autoManagedNetworkClientRegistry = autoManagedNetworkClientRegistry;
-    return autoManagedNetworkClientRegistry;
+    return (this.#autoManagedNetworkClientRegistry ??=
+      this.#createAutoManagedNetworkClientRegistry());
   }
 
   /**
-   * Constructs the registry of network clients based on the set of built-in
-   * networks as well as the custom networks in state.
+   * Constructs the registry of network clients based on the set of default
+   * and custom networks in state.
    *
    * @returns The network clients keyed by ID.
    */
   #createAutoManagedNetworkClientRegistry(): AutoManagedNetworkClientRegistry {
-    return [
-      ...this.#buildIdentifiedInfuraNetworkClientConfigurations(),
-      ...this.#buildIdentifiedCustomNetworkClientConfigurations(),
-    ].reduce(
+    const sortedChainIds = Object.keys(
+      this.state.networkConfigurationsByChainId,
+    )
+      .map(hexToBigInt)
+      .sort()
+      .map(toHex);
```

Comment:
nit: These are currently safe because `hexToBigInt`, `toHex` are unary, but might break if we add additional option arguments to these functions in the future. 

```suggestion
      .map((e) => hexToBigInt(e))
      .sort()
      .map((e) => toHex(e));
```

Reply:
Good shout. I'm okay with this — it seems unlikely that we would add new arguments any time soon, but this is an easy fix if that happens.